spectral_cube/tests/test_dask.py

```diff
@@ -239,6 +241,15 @@ def func(x, add=None):
                                    rslt3.filled_data[:].value)
 
 
+def test_cube_on_cube(data_adv):
+    # regression test for #782
+    cube = DaskSpectralCube.read(data_adv)
+
+    with patch.object(cube, '_cube_on_cube_operation') as mock:
+        cube * cube
+    mock.assert_called_once()
```

Comment:
maybe this is where we need an explicit deletion?
```suggestion
    mock.assert_called_once()
    del cube
```